pkg/apis/networking/types.go

```diff
@@ -699,3 +700,58 @@ type ClusterCIDRList struct {
 	// items is the list of ClusterCIDRs.
 	Items []ClusterCIDR
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IPAddress represents an IP used in a Kubernetes cluster, commonly associated to a Service ClusterIP.
+// An IP address can be represented in different formats, to guarantee the uniqueness of the IP,
+// the name of the object is the IP address in canonical format, four decimal digits separated
+// by dots suppressing leading zeros for IPv4 and the representation defined by RFC 5952 for IPv6.
+// Valid: 192.168.1.5 or 2001:db8::1
```

Comment:
should add the long form ipv6 here as well.

Reply:
so, we need the uniqueness on the object to allow to have concurrency on ClusterIP assignment without duplicates, if we allow multiple representations of the same IP we can have Services with duplicate ClusterIPs

Reply:
ack. Valid point.

I just wanted an example of a full IPv6 (without zero octets in the middle) which is still a unique value.